components/responsive-image/index.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+function ResponsiveImage( { src, alt, naturalWidth, naturalHeight } ) {
+	const imageStyle = {
+		paddingBottom: ( naturalHeight / naturalWidth * 100 ) + '%',
+	};
+	return (
+		<div className="components-responsive-image">
+			<div style={ imageStyle } />
+			<img src={ src } className="components-responsive-image__image" alt={ alt } />
```

Comment:
I tried to handle this, I've renamed `ResponsiveImage` as `ResponsiveWrapper` and it works by checking that the children is a single React Element and then cloning the element to add the class name. Does this sound good for you?